equinox/nn/_batch_norm.py

```diff
@@ -60,6 +60,7 @@ def __init__(
         channelwise_affine: bool = True,
         momentum: float = 0.99,
         inference: bool = False,
+        dtype: Num = jnp.float32,
```

Comment:
`Num` is only allowed to be used in conjunction with an array and shape, e.g. `Num[Array, "foo"]`.

Surprisingly, there actually isn't a valid dtype annotation that I know of. For now I'd just leave this unannotated.